src/app/who-did-this/who-did-this.component.ts

```diff
@@ -0,0 +1,28 @@
+import { Component, OnInit, OnDestroy } from '@angular/core';
+
+import { Observable, Subscription } from 'rxjs';
+
+import { WhoDidThisService } from './who-did-this.service';
+import { WhoDidThis } from '../shared/models/who-did-this.model';
+
+@Component({
+  selector: 'gp-who-did-this',
+  templateUrl: './who-did-this.component.html',
+  styleUrls: ['./who-did-this.component.scss']
+})
+export class WhoDidThisComponent implements OnInit, OnDestroy {
+  subscription: Subscription;
+  whoDidThis$: Observable<WhoDidThis[]>;
+
+  constructor(private whoDidThisService: WhoDidThisService) {}
+
+  ngOnInit() {
+    this.whoDidThis$ = this.whoDidThisService.getAll();
+  }
+
+  ngOnDestroy(): void {
+    if (this.subscription) {
+      this.subscription.unsubscribe();
+    }
+  }
```

Comment:
We don't need to implement OnDestroy because this subscription should be deleted.